doc/book/app_server/luajit_getmetrics.rst

```diff
@@ -0,0 +1,441 @@
+
+The other gc_*num values -- ``gc_cdatanum``, ``gc_tabnum``, ``gc_udatanum`` -- can be accessed
+in a similar way.
+Any of the gc_*num values can be useful when looking for memory leaks – the total 
+number of these objects should not grow nonstop.
+A more general way to look for memory leaks is to watch ``gc_total``.
+Also ``jit_mcode_size`` can be used to watch the amount of allocated memory for machine code traces.
+
+.. _luajit_getmetrics_example_2:
+
+Example with gc_allocated and and gc_freed
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+To track an application's effect on the garbage collector (less is better):
+
+.. code-block:: lua
+
+    function f()
+      for i = 1, 10 do collectgarbage("collect") end
+      local oldm = misc.getmetrics()
+      local newm = misc.getmetrics()
+      oldm = misc.getmetrics()
+      collectgarbage("collect")
+      newm = misc.getmetrics()
+      print("gc_allocated diff = " .. newm.gc_allocated - oldm.gc_allocated)
+      print("gc_freed diff = " .. newm.gc_freed - oldm.gc_freed)
+    end
+    f()
+
+The result will be: gc_allocated diff = 800, gc_freed diff = 800.
+This shows that ``local ... = getmetrics()`` itself causes memory allocation
+(because it is creating a table and assigning to it),
+and shows that when the name of a variable (in this case the ``oldm`` variable)
+is used again, that causes freeing.
+Ordinarily the freeing would not occur immediately, but
+``collectgarbage("collect")`` forces it to happen so we can see the effect.
+
+.. _luajit_getmetrics_example_3:
+
+Example with gc_allocated and a space optimization
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+To test whether optimizing for space is possible with tables:
+
+.. code-block:: lua
+
+    function f()
+      collectgarbage("collect")
+      local oldm = misc.getmetrics()
+      local t = {}
+      for i = 1, 513 do
+        t[i] = i
+      end
+      local newm = misc.getmetrics()
+      local diff = newm.gc_allocated - oldm.gc_allocated
+      print("diff = " .. diff)
+    end
+    f()
+
+The result will show that diff equals approximately 18000.
+
+Now see what happens if the table initialization is different:
+
+.. code-block:: lua
+
+    function f()
+      local table_new = require "table.new"
+      local oldm = misc.getmetrics()
+      local t = table_new(513, 0)
+      for i = 1, 513 do
+        t[i] = i
+      end
+      local newm = misc.getmetrics()
+      local diff = newm.gc_allocated - oldm.gc_allocated
+      print("diff = " .. diff)
+    end
+    f()
+
+The result will show that diff equals approximately 6000.
+
+.. _luajit_getmetrics_example_4:
+
+gc_steps_atomic and gc_steps_propagate
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The slope curves of gc_steps_* items can be used for tracking pressure on
+the garbage collector too.
+During long-running routines, gc_steps_* values will increase,
+but long times between ``gc_steps_atomic`` increases are a good sign.
```

Comment:
**** Okay, I added: "And, since ``gc_steps_atomic`` increases only once per garbage-collector cycle, it shows how many garbage-collector cycles have occurred."